objects/auth_factor.json

```diff
@@ -0,0 +1,96 @@
+{
+  "caption": "Authentication Factor",
+  "description": "Information on factors used to authenticate.",
+  "name": "authentication factor",
+  "extends": "object",
+  "attributes": {
+    "factor_device": {
+      "description": "Device used to complete the authentication request.",
+      "requirement": "recommended"
+    },
+    "factor_type": {
+      "enum": {
+        "0": {
+          "caption": "Unknown"
+        },
+        "1": {
+          "caption": "SMS",
+          "description": "A code sent to an end user's mobile device via SMS text message."
+        },
+        "2 question": {
+          "caption": "Security Question",
+          "description": "User answered a security question that only the user should know the answer to."
+        },
+        "3": {
+          "caption": "Phone Call",
+          "description": "System calls the user's registered phone number and requires the user to answer and provide a response."
+        },
+        "4": {
+          "caption": "Biometric",
+          "description": "Devices that verify identity based on user's physical identifiers, such as fingerprint scanners or retina scanners."
+        },
+        "5": {
+          "caption": "Push Notification",
+          "description": "Push notification sent to user's device and requires the user to acknowledge."
+        },
+        "6": {
+          "caption": "Hardware Token",
+          "description": "Physical devices that generate a code that can be used for authentication."
+        },
+        "7": {
+          "caption": "OTP",
+          "description": "Application generated a one-time password for use in authentication."
+        },
+        "8": {
+          "caption": "Email",
+          "description": "A code or link sent to a user's registered email address."
+        },
+        "9": {
+          "caption": "U2F",
+          "description": "Typically involves a hardware token which the user physically interacts with to authenticate."
+        },
+        "10": {
+          "caption": "WebAuthn",
+          "description": "Web based API allows users to register devices as factors."
+        },
+        "11": {
+          "caption": "Password",
+          "description": "Password that was previously created by the user is entered."
+        },
+        "99": {
+          "caption": "Other"
+        }
+      },
+      "requirement": "required"
+    },
+    "factor_question": {
+      "description": "Question provided for user to answer for question based authentication factor.",
+      "requirement": "optional"
+    },
+    "factor_telephony_number": {
+      "description": "Phone number for then telephony based MFA request.",
+      "requirement": "optional"
+    },
+    "factor_email": {
+      "description": "Email that an authentication was performed with",
+      "requirement": "optional"
+    },
+    "factor_provider": {
+      "description": "Provider of MFA Factor",
```

Comment:
`nit`: I wonder if we should use `provider` or `vendor`?  OCSF isn't consistent here:

Examples of "vendor_name":
* https://github.com/ocsf/ocsf-schema/blob/a94abe629672432781ea1973d6db13aab50c50a1/objects/product.json#L26-L29
* https://github.com/ocsf/ocsf-schema/blob/a94abe629672432781ea1973d6db13aab50c50a1/objects/vulnerability.json#L65-L68

Examples of "provider":
* https://github.com/ocsf/ocsf-schema/blob/a94abe629672432781ea1973d6db13aab50c50a1/objects/cloud.json#L16-L19
* https://github.com/ocsf/ocsf-schema/blob/a94abe629672432781ea1973d6db13aab50c50a1/objects/malware.json#L94-L97

I might ask in OCSF slack.

Reply:
Actually, as I wrote up more examples, it seems like this is just oriented towards the language used in the particular context.  CVEs for example are tied to "vendors" but we often talk about "cloud service providers".  My head goes towards saying Yubikey is a vendor but Duo is a service provider, I think because of how Yubikey is a peripheral device.  Provider is probably the best we can do.

Reply:
Sorry, what are CVEs? I tried to google but CVE is a popular acronym

Reply:
https://en.m.wikipedia.org/wiki/Common_Vulnerabilities_and_Exposures

Reply:
oh thank you!! maybe we can see if OCSF folks have better ideas than "provider"? but yes, agreed on not using Vendor based on all of this!